pandas/core/series.py

```diff
@@ -1601,6 +1601,7 @@ def __repr__(self) -> str:
         """
         Return a string representation for a particular Series.
         """
+        # pylint: disable=invalid-repr-returned
         repr_params = fmt.get_series_repr_params()
         return self.to_string(**repr_params)
```

Comment:
This seems to solve the issue. Should we go for this? 

```suggestion
        return str(self.to_string(**repr_params)) # GH#49025
```

Also, `self.to_string` method has a validation to ensure  only str values are returned. 

https://github.com/pandas-dev/pandas/blob/2a0e3fe89ba786e54f1cf7c46586de1a6151c438/pandas/core/series.py#L1700-L1709